main.go

```diff
@@ -873,7 +879,12 @@ func buildAndRun(pkgName string, config *compileopts.Config, stdout io.Writer, c
 	} else {
 		cmd = exec.Command(name, args...)
 	}
-	cmd.Env = env
+	if len(extraCmdEnv) > 0 {
+		cmd.Env = append(cmd.Env, env...)
+		cmd.Env = append(cmd.Env, extraCmdEnv...)
+	} else {
+		cmd.Env = env
+	}
```

Comment:
I'm not sure this is more readable, but at least it does basically the same thing without needing an if/else.

```suggestion
	cmd.Env = append([]string, env...)
	cmd.Env = append(cmd.Env, extraCmdEnv)
```

To me it's more readable, not sure about other people :)

Reply:
I think I was trying to avoid the extra allocation the default case where we didn't have it, or potentially modifying the `env` array.   But since this is the compiler it probably doesn't matter as much.

Reply:
It happens only once per run (in the compiler!), so this is really not a place to worry about performance. So then, the thing that I find important the most is readability :)

Anyway, LGTM either way, you can fix it if you think it's worth it.